### DISCUSSION TITLE
Displaying dynamic text in a scene

### DISCUSSION THREAD

**Discussion by wpinaud** [google/filament#3551] [Q&A] [answered]
Hi Filament community,

I'm trying to display dynamic text in a 3D scene. 
The scene is a digital clock, and I would like to display the current time on it.

I've been exploring if Filament has a specific renderable for Text assets but it does not seem to be the case.
Now I am looking at shaders which could render text based on a texture with numbers by mapping each number with his UVs.

But I'm wondering if there is a more performant/simpler way of achieving it,
All propositions, ideas would be welcome.

**Comment by romainguy** [comment, marked as answer]
What you are describing is effectively the simplest/cheapest way of doing it. For improved quality at different scales, signed distance fields are commonly used as well. Then there are more advanced but higher quality solutions like https://sluglibrary.com/

**Reply by wpinaud** [reply]
Thank you for your confirmation, I'll dig more into it!
